### ISSUE TITLE
design settings. text size, visibility etc

### ISSUE THREAD

**Issue by pioupus** [pioupus/kicadLibCreator#11]

**Comment by Gasman2014** [comment]
Components have very small default annotation sizes. e.g. a capacitor I placed had its value displayed as 100nF in 0.254mm high text - I though there was a fleck of dust on my screen at first! Most component I have seen seem to have 1.27mm as a standard size.

Also default text font colour for values seems to be magenta - not the default aqua green.

**Comment by pioupus** [comment]
Now the 100nF is stored in the Field Display value and not as the value itself. Therefore unfortunatly I can't do anything about the colors. Hopefully the Kicad guys will integrate color settings.

**Comment by Gasman2014** [comment]
Shame - although not a major issue - I guess on a different level, identifies on the schematic that the part is already fully defined. 

Still getting very small default text sizes.

**Comment by pioupus** [comment]
Have you tried do adjust the design settings in "Extras/Design settings"? You also have to start Eeschema since as far as I know it loads its libraries on start up. (Also something I should ask the Kicad developers)

**Comment by Gasman2014** [comment]
How did I not see that before?!

That works - yes, can now adjust settings and position. Still having some issues with this though. For instance the default Capacitor 

```
F0 "C" 25 100 50 H V L CNN
F1 "C" 25 -100 50 H V L CNN
F2 "" 38 -150 50 H V C CNN
```
and the new footprint (with no adjustments of position)
```
F0 "C" 0 0 1 H I L TNN ""
F1 "C0805C104K5RACTU" 0 0 1 H I L TNN ""
F2 "C_0805" 0 0 1 H I L TNN "footprint"
...
F7 "100 nF" 0 0 1 H V L TNN "DisplayValue"
```

and the standard resistor

```
DEF R R 0 0 N Y 1 F N
F0 "R" 80 0 50 V V C CNN
F1 "R" 0 0 50 V V C CNN
F2 "" -70 0 50 V V C CNN
F3 "" 0 0 50 H V C CNN
```

compared to the new version.

```
DEF RC1206JR-076K8L R 0 3 N Y 1 F N
F0 "R" 80 0 40 V V C CNN ""
F1 "RC1206JR-076K8L" 0 0 40 H I L BNN ""
F2 "R_1206" 0 0 40 H I L BNN "footprint"
F3 "/Users/johnpateman/Dropbox/Components/Datasheets/resistor_smd/yageo/rc1206jr-076k8l.pdf" 0 0 40 H I L BNN "datasheet"
```

![screen shot 2016-12-15 at 11 17 45](https://cloud.githubusercontent.com/assets/8401593/21222303/120138a0-c2b9-11e6-90d0-c7894a0cf74d.png)

New on left and default on right. I am not sure that the orientation and positioning is being respected - I would have thought that the RefDes and Value position and orientation should, by default, be copied from the source device. 

Or have I missed something here?

**Comment by pioupus** [comment]
does it work now?

**Comment by Gasman2014** [comment]
Yes, that is much better! Works nicely now. Need to try it with a wider range of components to be sure. Like the use of the slightly smaller fonts too.

**Comment by Gasman2014** [comment]
Yes, much better:

![screen shot 2016-12-17 at 11 42 28](https://cloud.githubusercontent.com/assets/8401593/21286515/12d0bcce-c44e-11e6-8ed7-2de857899385.png)
New above, old from 'device.lib' below.

Although the LED is slightly strange as it has the cathode and anode marked in the new library version - but when I look at the DRAW parameters the default and the new one seem identical. Not a significant problem but not sure why it is so.

Default (no visible KA marks)

```
# LED
#
DEF LED D 0 40 Y N 1 F N
F0 "D" 0 100 50 H V C CNN
F1 "LED" 0 -100 50 H V C CNN
F2 "" 0 0 50 H V C CNN
F3 "" 0 0 50 H V C CNN
$FPLIST
 LED-3MM
 LED-5MM
 LED-10MM
 LED-0603
 LED-0805
 LED-1206
 LEDV
$ENDFPLIST
DRAW
P 2 0 1 0 -50 50 -50 -50 N
P 3 0 1 0 -80 -25 -125 -65 -120 -40 N
P 3 0 1 0 -65 -40 -110 -80 -105 -55 N
P 3 0 1 0 50 50 -50 0 50 -50 F
X K 1 -200 0 150 R 40 40 1 1 P
X A 2 200 0 150 L 40 40 1 1 P
ENDDRAW
```

And the custom version (K & A marked)

```
DEF SML-LX1206GC-TR D 0 3 N Y 1 F N
F0 "D" 0 100 40 H V C CNN ""
F1 "SML-LX1206GC-TR" 0 0 40 H I L BNN ""
F2 "LED_1206" 0 0 40 H I L BNN "footprint"
F3 "/*****/Components/Datasheets/led/lumex/sml-lx1206gc-tr.pdf" 0 0 40 H I L BNN "datasheet"
F4 "1481841632573" 0 0 40 H I L BNN "key"
F5 "SML-LX1206GC-TR" 0 0 40 H I L BNN "man"
F6 "Lumex" 0 0 40 H I L BNN "manufacturer"
F7 "Green" 0 -100 40 H V C CNN "DisplayValue"
DRAW
P 2 0 1 0 -50 50 -50 -50 N
P 3 0 1 0 -80 -25 -125 -65 -120 -40 N
P 3 0 1 0 -65 -40 -110 -80 -105 -55 N
P 3 0 1 0 50 50 -50 0 50 -50 F
X K 1 -200 0 150 R 40 40 1 1 P
X A 2 200 0 150 L 40 40 1 1 P
ENDDRAW
```

**Comment by Gasman2014** [comment]
I can see what it is now - 

Default definition is
DEF LED D 0 40 **Y** **N** 1 F N

Custom is 
DEF SML-LX1206GC-TR D 0 3 **N** **Y** 1 F N

According to the library docs, I see that Parameter 5 is Pin Number Visibility and Parameter 6 is Pin Name Visibility. These seem to be reversed here. Not a fatal error but bit confusing at first.

**Comment by pioupus** [comment]
I think this is fixed now. You may have to check your design settings (that "overwrite pin style checkbox)
